src/powfacpy/base_interface.py

```diff
@@ -933,6 +933,47 @@ def handle_path(path):
         return path[1:] 
     except(TypeError):
       raise TypeError("Path must be of type string.")
+  
+  @staticmethod
+  def replace_outside_of_strings_in_a_string(string: str, replacements: dict):
+    """This method replaces parts of a string but only in the sections
+    of the original string that are outside of strings. This is best
+    explained by example:
+    "p HV load >= 2 and 'This is a string inside the string'"
+    In this string the part 'This is a string inside the string'
+    is a string inside the original string and is no replacements are
+    made in this part.
+
+    Arguments:
+      - string: The string that will be adjusted
+      - replacements: key-value pairs of matching strings and their replacement
+    """
+    string = PFStringManipulation.split_but_keep_delimiter(string,"'")
```

Comment:
maybe make the choice of inside and outside string delimiter flexible, since people do this differently by preference.
something like:

if "'" in string:
  inside_string_delimiter = "'"
else:
  inside_string_delimiter = '"'
string = PFStringManipulation.split_but_keep_delimiter(string, inside_string_delimiter)

Reply:
Ok, can be adjusted in future

Reply:
I added this now already